### DISCUSSION TITLE
Ability to toggle all coclist entries and send them to qflist

### DISCUSSION THREAD

**Discussion by asmodeus812** [neoclide/coc.nvim#4953] [Q&A]
Hi, i am looking at a way to implement something like toggleAll, but looking at the api i am not seeing any way to extract the full list contents, further more i was wondering if it is possible to make the default action - <CR>, send all toggled items to qflist or if there were no toggled items, simply execute whatever the default action is for a given list. Thanks !

**Comment by fannheyward** [comment]
Try `nvim-bqf`, supports diagnostics list, not all types list.

**Reply by asmodeus812** [reply]
I was looking for a solution to enable select all + sending to qf list for any type of list, outline, symbols, diagnostics etc. Not sure why this relates to nvim-bqf but not looking to use another plugin for this purpose atm.

**Reply by asmodeus812** [reply]
I did find out about the selectall action on a CocList, so that kind of solved the selectall issue, and somewhat solved the send to qf or default action like that - but would need this to get merged in - https://github.com/neoclide/coc.nvim/pull/4954

```viml
function! Listaction(ctx)
    " do execute quickfix or default action
    return v:lua.list_default_action(a:ctx)
endfunction
```

```lua
function _G.list_default_action(ctx)
        if ctx and #ctx.targets > 1 then
            -- send to qflist
            return "quickfix"
        else
            -- default
            return nil
        end
    end
```
